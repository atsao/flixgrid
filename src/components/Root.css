body {
  font-family: Helvetica Neue, Arial, sans-serif;
  font-size: 14px;
}

.wrapper {
  margin: 0 auto;
  width: 90%
}

@media (max-width: 568px) {
  .wrapper {
    overflow: scroll;
    min-width: 48em;
  }
}
