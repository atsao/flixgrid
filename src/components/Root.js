import React, { Component } from 'react';
import PropTypes from 'prop-types';

import styles from '../assets/css/index.css';
import data from '../data.json';

import Flixgrid from './Flixgrid';

class Root extends Component {
  constructor(props) {
    super(props);

    this.state = {
      data: [],
    };
  }

  componentDidMount() {
    setTimeout(
      () => {
        this.setState({
          data,
        });
      },
      1000
    );
  }
  render() {
    const { data } = this.state;
    const columns = [
      {
        dataKey: 'title',
        display: 'Title',
        render: data => <strong>{data}</strong>,
      },
      {
        dataKey: 'releaseYear',
        display: 'Release Year',
      },
      {
        dataKey: 'rating',
        display: 'Rating',
      },
      {
        dataKey: 'genre',
        display: 'Genre',
      },
    ];
    return (
      <div className={styles['wrapper']}>
        <Flixgrid data={data} columns={columns} showHeaders />
      </div>
    );
  }
}

Root.propTypes = {
  children: PropTypes.node,
};

export default Root;
