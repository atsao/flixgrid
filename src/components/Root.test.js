import React from 'react';
import renderer from 'react-test-renderer';

import Root from './Root';

it('renders a Root component', () => {
  const tree = renderer.create(<Root />);
  expect(tree).toMatchSnapshot();
});
