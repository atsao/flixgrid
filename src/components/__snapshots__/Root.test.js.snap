// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Root component 1`] = `
<div
  className="wrapper"
>
  <table
    className="flixgrid-table"
  >
    <thead>
      <tr
        className="false flixgrid-thead"
      >
        <td>
          Title
           
          <button
            className="sort-ascending btn btn-ghost "
            disabled={false}
            onClick={[Function]}
          >
            ▲
          </button>
           
          <button
            className="sort-descending btn btn-ghost "
            disabled={false}
            onClick={[Function]}
          >
            ▼
          </button>
           
          <input
            className="filter"
            onChange={[Function]}
            placeholder="Filter on title"
            type="text"
          />
        </td>
        <td>
          Release Year
           
          <button
            className="sort-ascending btn btn-ghost "
            disabled={false}
            onClick={[Function]}
          >
            ▲
          </button>
           
          <button
            className="sort-descending btn btn-ghost "
            disabled={false}
            onClick={[Function]}
          >
            ▼
          </button>
           
          <input
            className="filter"
            onChange={[Function]}
            placeholder="Filter on release year"
            type="text"
          />
        </td>
        <td>
          Rating
           
          <button
            className="sort-ascending btn btn-ghost "
            disabled={false}
            onClick={[Function]}
          >
            ▲
          </button>
           
          <button
            className="sort-descending btn btn-ghost "
            disabled={false}
            onClick={[Function]}
          >
            ▼
          </button>
           
          <input
            className="filter"
            onChange={[Function]}
            placeholder="Filter on rating"
            type="text"
          />
        </td>
        <td>
          Genre
           
          <button
            className="sort-ascending btn btn-ghost "
            disabled={false}
            onClick={[Function]}
          >
            ▲
          </button>
           
          <button
            className="sort-descending btn btn-ghost "
            disabled={false}
            onClick={[Function]}
          >
            ▼
          </button>
           
          <input
            className="filter"
            onChange={[Function]}
            placeholder="Filter on genre"
            type="text"
          />
        </td>
      </tr>
    </thead>
    <tbody>
      <tr
        className="flixgrid-tr "
      >
        <td
          className="flixgrid-td-empty"
          colSpan={4}
        >
          No data
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
